cishouseholds/edit.py

```diff
@@ -10,6 +11,36 @@
 from cishouseholds.pyspark_utils import get_or_create_spark_session
 
 
+def update_household_count(
+    df: DataFrame,
+    column_name_to_update: str,
+    infant_column_pattern: str,
+    infant_column_pattern_with_exceptions: str,
+    participant_column_pattern: str,
+    non_consented_count: str,
+):
+    """ """
+    infant_columns = [x for x in df.columns if re.match(infant_column_pattern, x)]
+    infant_columns_with_exceptions = [x for x in df.columns if re.match(infant_column_pattern_with_exceptions, x)]
+    participant_columns = [x for x in df.columns if re.match(participant_column_pattern, x)]
+
+    df = df.withColumn("INFANT_ARRAY", F.array([F.when(F.col(col).isNull(), 0).otherwise(1) for col in infant_columns]))
```

Comment:
Please change all of these intermediate columns to objects, rather than creating columns that need dropping later